libraries/slave_windows.rb

```diff
@@ -128,6 +142,22 @@ def slave_compat_xml
       @slave_compat_xml
     end
 
+    
+    def user_hash
```

Comment:
agreed.  Problem is this is [WinSW] ( https://github.com/kohsuke/winsw) related which  might have different requierments.  But with that said, the regex and split I would say makes sense to be common.